src/lib/transaction/Transaction.svelte

```diff
@@ -17,33 +17,42 @@
         const keyPair = await getStellarKeypair(privateKey);
         return keyPair;
     }
-
     const keyPair = getKeyPair();
 
     let tx: Transaction;
     let operationComponents: typeof OperationComponentTypes[] = [];
-
     const isValidXdr = writable(false);
-    const xdrValue = location.search.substring(5);
+
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const xdrValue = urlParams.get('xdr')?.replace(/\s/g, '+');
+    const description = urlParams.get('description');
 
     try {
-        $isValidXdr = xdr.TransactionEnvelope.validateXDR(xdrValue, 'base64');
-        tx = new Transaction(xdrValue, import.meta.env.VITE_HORIZON_NETWORK_PASSPHRASE);
+        $isValidXdr = xdr.TransactionEnvelope.validateXDR(xdrValue!, 'base64');
+        tx = new Transaction(xdrValue!, import.meta.env.VITE_HORIZON_NETWORK_PASSPHRASE);
```

Comment:
why not move the `try` condition to the very top and avoid marking this as `!`?  Let it fail on the `replace` if the `xdr` is null?